src/core/streams/actions/morph.js

```diff
@@ -26,7 +26,10 @@ function beforeNodeRemoved(node) {
 }
 
 function beforeNodeMorphed(target, newElement) {
-  if (target instanceof HTMLElement && !target.hasAttribute("data-turbo-permanent")) {
+  if (!(target instanceof HTMLElement)) {
+    return
+  }
+  if (!target.hasAttribute("data-turbo-permanent")) {
```

Comment:
The issue with this implementation is that the `turbo:before-morph-element` event will not be dispatched for text morphs, instead I would change the if condition to a logical or that way events will be consistently dispatched
```suggestion
  if () {
    return
  }
  if (!(target instanceof HTMLElement) || !target.hasAttribute("data-turbo-permanent")) {
```